src/lib/web3-utils.js

```diff
@@ -196,3 +196,18 @@ export async function checkValidEthNode(uri) {
 
   return true
 }
+
+export async function signMessage(wallet, message) {
+  let signHash
+  let error = false
+  let errorMsg
+
+  try {
+    signHash = await wallet.ethers.getSigner().signMessage(message)
```

Comment:
I know we discussed this earlier, but on second thought, what was the motivation behind catching the error here and returning it rather than letting it bubble up to the user?